src/Service/NotificationChannelFactory.php

```diff
@@ -0,0 +1,47 @@
+<?php
+
+namespace App\Service;
+
+use Psr\Container\ContainerInterface;
+use Psr\Log\LoggerInterface;
+use Symfony\Component\HttpFoundation\RequestStack;
+use Symfony\Contracts\Service\ServiceSubscriberInterface;
+
+class NotificationChannelFactory implements ServiceSubscriberInterface
+{
+    private $requestStack;
+    private $locator;
+    private $logger;
+
+    public function __construct(
+        RequestStack $requestStack,
+        ContainerInterface $locator,
+        LoggerInterface $logger
+    ) {
+        $this->requestStack = $requestStack;
+        $this->locator = $locator;
+        $this->logger = $logger;
+    }
+
+    public static function getSubscribedServices(): array
+    {
+        return [
+            'EmailNotificationChannel' => EmailNotificationChannel::class,
+            'FileLoggerNotificationChannel' => FileLoggerNotificationChannel::class,
+            'TelegramNotificationChannel' => TelegramNotificationChannel::class,
+        ];
+    }
+
+    public function create(): ?NotificationChannelInterface
+    {
+        $request = $this->requestStack->getCurrentRequest();
+        $currentChannel = $request->get('channel');
```

Comment:
it's a bad practice to depend on request in services, as this mixes application layers ad disallows using these services in other contexts, like CLI, where there is no request. Consider passing channel as an argument to the factory method